### ISSUE TITLE
Modular Coding

### ISSUE THREAD

**Issue by susantkumarsahoo** [susantkumarsahoo/StudentPerformancePrediction-End-To-End#19]
# important data validation technique names specifically for Machine Learning (ML) and Deep Learning (DL)

# 🧩 1. Schema & Structure Validation

Column Name Validation

Data Type Validation

Missing Column Check

Extra Column Detection

Row Count Validation

Primary Key Uniqueness Check

Schema Consistency Validation

# 🔍 2. Data Quality & Integrity Validation

Missing Value Validation

Duplicate Record Detection

Outlier Detection

Range Validation

Categorical Value Validation

Business Rule Validation

Logical Consistency Check

Cross-Field Consistency Check

Data Leakage Check

Target Column Integrity Check

Data Drift Validation

Feature Drift Validation

Distribution Stability Check

# 📊 3. Statistical Validation

Mean-Std Deviation Validation

Z-Score Validation

IQR Validation

Variance Threshold Check

Correlation Validation

Covariance Validation

Skewness-Kurtosis Check

Population Stability Index (PSI)

Feature Importance Stability Check

Statistical Summary Validation

# 🧠 4. Model Input Data Validation

Train-Test Feature Alignment Check

Scaling/Normalization Validation

One-Hot Encoding Consistency Check

Categorical Encoding Validation

Feature Range Validation (post-scaling)

Target Leakage Detection

Missing Value Ratio Validation

Feature Distribution Comparison

Feature Consistency Check

Feature Type Validation

# 🕒 5. Time Series Validation

Timestamp Order Validation

Missing Timestamp Check

Duplicate Timestamp Detection

Time Interval Consistency Check

Seasonality Pattern Validation

Frequency Validation

Lag Feature Validation

Rolling Window Validation

Temporal Leakage Detection

Time Gap Detection

# 💬 6. Text / NLP Data Validation

Text Encoding Validation

Empty Text Detection

Text Length Validation

Token Count Validation

Language Detection

Non-Printable Character Check

Stopword Ratio Validation

Punctuation Ratio Validation

Text Normalization Validation

Vocabulary Consistency Validation

# ⚙️ 7. Feature Engineering Validation

Feature Creation Validation

Feature Transformation Validation

Derived Feature Sanity Check

Log/Power Transformation Validation

Polynomial Feature Validation

Interaction Feature Validation

Encoding Mapping Validation

Feature Scaling Range Check

# 🧰 8. Automation & Framework Validation

Great Expectations Validation

Pandera Schema Validation

Deepchecks Data Validation

TensorFlow Data Validation (TFDV)

Pydantic Data Model Validation

# 📈 9. Model Readiness Validation

Input-Output Shape Validation

Batch Size Consistency Check

NaN / Inf Value Detection

Dataset Splitting Validation

Data Shuffling Validation

Target Distribution Validation

Class Imbalance Check

Sampling Strategy Validation

# 🚀 10. Production & Post-Training Validation

Data Pipeline Drift Detection

Real-Time Data Validation

Feature Store Consistency Validation

Model Input Schema Drift

Model Serving Data Validation

Prediction Range Validation

Monitoring Feature Stability

Data Refresh Validation

Retraining Trigger Validation
